Handle missing config file error

BenCapper opened the pull request:
Fixes: #139 

- Return a more elegant error message when no config file is passed to the engine

Signed-off-by: Ben Capper <BenCapper19@gmail.com>

BenCapper committed:
Handle missing config file in setup command

Return a more elegant error message.

Co-authored-by: Ben Capper <BenCapper19@gmail.com>
Co-authored-by: Akashdeep Dhar <akashdeep.dhar@gmail.com>
Co-authored-by: Nils Philippsen <nils@redhat.com>
Signed-off-by: Akashdeep Dhar <akashdeep.dhar@gmail.com>
Signed-off-by: Nils Philippsen <nils@redhat.com>

gridhead reviewed (changes requested):
Can be a bit more elegant, I think ;P

```
(venv) [t0xic0der@fedorable duffy]$ duffy setup-db
```
```
KeyError: 'database'

The above exception was the direct cause of the following exception:

duffy.exceptions.DuffyConfigurationError: database

During handling of the above exception, another exception occurred:

duffy.exceptions.DuffyConfigurationError: Missing Configuration File
```

nphilipp committed:
Fix exception raised in test_shell()

DuffyConfigurationError only needs the problematic configuration key as
an argument, the message will be added in the (tested) CLI command.

Signed-off-by: Nils Philippsen <nils@redhat.com>

gridhead reviewed (approved):
LGTM.

Merging.

```
(venv) [t0xic0der@fedorable duffy]$ duffy serve
 * Starting Duffy...
 * Host address : 127.0.0.1
 * Port number  : 8080
 * Log level    : warning
 * Serving API docs on http://127.0.0.1:8080/docs
ERROR:duffy.cli:Configuration key missing or wrong: database
(venv) [t0xic0der@fedorable duffy]$ duffy setup-db
ERROR:duffy.cli:Configuration key missing or wrong: database
(venv) [t0xic0der@fedorable duffy]$ duffy -c duffy/some-config-file-which-doesnt-exist.yml setup-db
ERROR:duffy.cli:Configuration key missing or wrong: database
(venv) [t0xic0der@fedorable duffy]$ duffy -c duffy/some-config-file-which-doesnt-exist.yml serve
 * Starting Duffy...
 * Host address : 127.0.0.1
 * Port number  : 8080
 * Log level    : warning
 * Serving API docs on http://127.0.0.1:8080/docs
ERROR:duffy.cli:Configuration key missing or wrong: database
```
